### ISSUE TITLE
unrecognized import path "vbom.ml/util"

### ISSUE THREAD

**Issue by asalkeld** [metal3-io/baremetal-operator#651]
I can't build on Fedora 32, go1.14.9 as I keep getting

```
make build
LDFLAGS=-X github.com/metal3-io/baremetal-operator/pkg/version.Raw=capm3-v0.4.0-77-g44bcabd59c0b20b9ae87800f2bc4afcc457e07f6 -X github.com/metal3-io/baremetal-operator/pkg/version.Commit=44bcabd59c0b20b9ae87800f2bc4afcc457e07f6
go build -ldflags "-X github.com/metal3-io/baremetal-operator/pkg/version.Raw=capm3-v0.4.0-77-g44bcabd59c0b20b9ae87800f2bc4afcc457e07f6 -X github.com/metal3-io/baremetal-operator/pkg/version.Commit=44bcabd59c0b20b9ae87800f2bc4afcc457e07f6" -o build/_output/bin/baremetal-operator cmd/manager/main.go
go: github.com/operator-framework/operator-registry@v1.13.8 requires
	k8s.io/kubectl@v0.18.0 requires
	vbom.ml/util@v0.0.0-20160121211510-db5cfe13f5cc: unrecognized import path "vbom.ml/util": https fetch: Get "https://vbom.ml/util?go-get=1": dial tcp: lookup vbom.ml on 10.0.0.138:53: no such host
make: *** [Makefile:178: build] Error 1
```

looks like this: https://github.com/kubernetes/kubectl/issues/925
I even totally deleted my whole go tree and it still resulted in the same error.

**Comment by stbenjam** [comment]
I think the problem is the default GOPROXY setting on Fedora, if this is what @honza and @mhrivnak were talking about the other day. 

It should be GOPROXY="https://proxy.golang.org,direct"

**Comment by mhrivnak** [comment]
There is a good explanation starting here, mostly from Joe's analysis: https://github.com/operator-framework/operator-sdk/issues/3859#issuecomment-697771521

TL;DR the "problem" is not GOPROXY. The problem is that a package used a vanity domain for its import path, then its site went down, and now nobody can access that module unless they access it from a cache that happens to have a copy. Fedora defaults to not using Google's proxy (proxy.golang.org) due AIUI to privacy and transparency reasons, so this problem is more likely to be experienced on Fedora than other platforms.

Many projects are now experiencing this problem indirectly because kubectl uses the library: https://github.com/kubernetes/kubectl/issues/925

As a workaround, if you don't mind using Google's privately-run cache, this would probably work:

```
GOPROXY="https://proxy.golang.org,direct" make build
```

**Comment by asalkeld** [comment]
thanks for the great explanation! Should I close or leave open for a while for others to find?

**Comment by honza** [comment]
I guess this can be added to the list of arguments in favour of vendoring our dependencies.

**Comment by mhrivnak** [comment]
> Should I close or leave open for a while for others to find?

I think it's worth trying to add a `replace` statement in go.mod rather than depend on the proxy, so I'd leave this open for now.

**Comment by mhrivnak** [comment]
I reached out to the maintainer, and he was able to get the redirect up and running again for vbom.ml/util. This problem should be resolved for now, and the new import path should be picked up naturally whenever operator-sdk rebases onto k8s 1.20, and BMO picks up that change.
